src/init/starship.xsh

```diff
@@ -0,0 +1,16 @@
+import uuid
+
+
+def starship_prompt():
+    last_cmd = __xonsh__.history[-1] if __xonsh__.history else None
+    status = last_cmd.rtn if last_cmd else 0
+    # I believe this is equivalent to xonsh.jobs.get_next_job_number() for our purposes,
+    # but we can't use that function because of https://gitter.im/xonsh/xonsh?at=60e8832d82dd9050f5e0c96a
+    jobs = sum(1 for job in __xonsh__.all_jobs.values() if job['obj'] and job['obj'].poll() is None)
+    duration = round((last_cmd.ts[1] - last_cmd.ts[0]) * 1000) if last_cmd else 0
+    return $(::STARSHIP:: prompt --status=@(status) --jobs=@(jobs) --cmd-duration=@(duration))
```

Comment:
Now that `$STARSHIP_SESSION_KEY` works I've found another issue:
Log messages (stderr) from starship don't get displayed (invalid config file). Maybe it needs a workaround like this for [PowerShell](https://github.com/starship/starship/blob/master/src/init/starship.ps1#L56-L62)?

Reply:
seems to work for me when using https://github.com/anki-code/xontrib-prompt-starship latest and 

```
@ xonsh --version
xonsh/0.9.27
@ python -V
Python 3.9.2
@ starship --version
starship 0.55.0
tag:v0.55.0
commit_hash:09b12a52
build_time:2021-06-20 18:31:06
build_env:rustc 1.53.0 (53cb7b09b 2021-06-17)
```

![image](https://user-images.githubusercontent.com/1596188/125705937-92241e51-c42f-4c6b-98a2-74cc5a3075d0.png)
![image](https://user-images.githubusercontent.com/1596188/125706784-32fdc2ef-f1a6-41c8-85d5-832e71a23f18.png)

Reply:
Good catch, @davidkna.

This was fixed upstream in xonsh just three weeks ago: https://github.com/xonsh/xonsh/pull/4336

But since no released versions of xonsh contain that fix, we should probably work around it here. I've added a workaround: https://github.com/starship/starship/pull/2807/commits/1e0bc8d6a3ccb205b1f4069d29776a8f44285a75